.github/workflows/action-ci.yml

```diff
@@ -0,0 +1,160 @@
+
+  build:
+    needs: defineBuilds
+    name: Build
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        platform: ${{ fromJSON(needs.defineBuilds.outputs.platformList) }}
+    env:
+      buildVersion: ${{ fromJSON(needs.defineBuilds.outputs.version) }}
+    steps:
+    - uses: actions/checkout@v4
+    - uses: microsoft/setup-msbuild@v2
+      name: Setup msbuild
+    - uses: nuget/setup-nuget@v2
+      name: Use nuget 6.x
+      with:
+        nuget-version: '6.x'
+    - run: nuget restore ./src/Calculator.sln
+      name: Restore nuget dependencies
+    - run: |
+        ./build/scripts/UpdateAppxManifestVersion.ps1 `
+          -AppxManifest ./src/Calculator/Package.appxmanifest `
+          -Version ${{ env.buildVersion }}
+      shell: pwsh
+      name: Set version number in AppxManifest
+    - run: |
+        msbuild ./src/Calculator.sln `
+          -bl:${{ github.workspace }}/output/Calculator.binlog `
+          -p:OutDir=${{ github.workspace }}\output\ `
+          -p:Platform=${{ matrix.platform }} `
+          -p:Configuration=Release `
+          -p:GenerateProjectSpecificOutputFolder=true `
+          -p:Version=${{ env.buildVersion }} `
+          -maxCpuCount `
+          -t:Publish `
+          -p:PublishDir=${{ github.workspace }}\output\publish\
+      shell: pwsh
+      name: Build calculator.sln
+    - uses: actions/upload-artifact@v4
+      name: Upload build outputs
+      with:
+        name: Build-${{ matrix.platform }}
+        path: ${{ github.workspace }}/output
+
+  unitTests:
+    needs: [defineBuilds, build]
+    runs-on: windows-latest
+    name: Run unit tests
+    strategy:
+      matrix:
+        platform: ${{ fromJSON(needs.defineBuilds.outputs.unitTestPlatformList) }}
+    env:
+      testDir: ${{ github.workspace }}/download/CalculatorUnitTests/AppPackages/CalculatorUnitTests_Test
+    steps:
+    - uses: actions/download-artifact@v4
+      name: Download build outputs
+      with:
+        name: Build-${{ matrix.platform }}
+        path: ${{ github.workspace }}/download
+    - run: |
+        ${{ env.testDir }}/Add-AppDevPackage.ps1 `
+          -CertificatePath ${{ env.testDir }}/CalculatorUnitTests.cer `
+          -Force
+      shell: pwsh
+      name: Install test certificate
+    - uses: ilammy/msvc-dev-cmd@v1  # this is a workaround because microsoft/vstest-action is broken.
+      name: Setup dev tools
+    - run: vstest.console.exe ${{ env.testDir }}/CalculatorUnitTests.msix /Platform:${{ matrix.platform }}
+      name: Run unit tests
+        
+  uiTests:
+    needs: build
+    runs-on: windows-latest
+    name: Run UI tests (x64)
+    env:
+      appDir: ${{ github.workspace }}/download/Calculator/AppPackages/Calculator*_Test
+      pubDir: ${{ github.workspace }}/download/publish
+    steps:
+    - uses: actions/download-artifact@v4
+      name: Download build outputs
+      with:
+        name: Build-x64
+        path: ${{ github.workspace }}/download
+    - run: |
```

Comment:
Q: Do we need to turn off animation effects like existing CI?

Reply:
Good question. The image for ui testing is now Windows Server which is different from the Windows 11 image that ADO is using. I haven't found the evidence to prove that the Animation Effects are off on `windows-latest` but I believe it should be off by default.
Since the UI testing is successful and the time internal has no big difference, I think it's fine to skip this step until we found some problem.

Reply:
Yeah, looks good to me😆